apps/nextjs/src/components/AppComponents/Chat/chat-layout.tsx

```diff
@@ -13,7 +13,11 @@ export interface ChatLayoutProps {
 }
 
 export const ChatLayout = ({ className }: Readonly<ChatLayoutProps>) => {
-  const { isLoading, lessonPlan, messages } = useLessonChat();
+  const { lessonPlan, messages } = useLessonChat();
+  const stableMessages = useChatStore((state) => state.stableMessages);
+  const streamingMessage = useChatStore((state) => state.streamingMessage);
+  const messagesFromStore = [...stableMessages, streamingMessage];
```

Comment:
This is going to be a different array object each render, I think it needs to be in a useMemo

It's ok to group them into one list as a drop in replacement for now, but let's put a TODO in here to remind us to split the rendering of stableMessages and streamingMessage so we don't need to rerender everything when just the streamingMessage is changing